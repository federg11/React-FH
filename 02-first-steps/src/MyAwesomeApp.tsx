
export const MyAwesomeApp = () => {

    const name = 'Federico';
    const lastname = "Galup";
  return (
   <>
     <h1 data-testid="first-name-title">{name}</h1>
    <h3>{lastname}</h3>
   </>
)
}
