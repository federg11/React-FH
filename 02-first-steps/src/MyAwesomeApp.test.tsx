import { describe, expect, test } from "vitest";
import {render, screen} from '@testing-library/react'

import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {
    const {container} =   render( <MyAwesomeApp />);
    
        const h1 = container.querySelector('h1');
         const h3 = container.querySelector('h3');
        expect(h1?.innerHTML).toContain('Federico');
        expect(h3?.innerHTML).toContain('Galup')
    });

    test('should render firstName and lastName - screen', () => {
       render( <MyAwesomeApp />);
        screen.debug();

        // const h1 = screen.getByRole('heading', {
        //     level: 1
        // })

        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Federico')
    })

    test('should match snapshot', () => {
      const {container} = render(<MyAwesomeApp />);

      expect(container).toMatchSnapshot();
    })
})