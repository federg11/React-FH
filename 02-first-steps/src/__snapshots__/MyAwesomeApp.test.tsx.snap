// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <h1
    data-testid="first-name-title"
  >
    Federico
  </h1>
  <h3>
    Galup
  </h3>
</div>
`;
